File: Base/Desktop/Allors/Client/Custom/Person.ts
namespace Allors.Domain
{
    export interface Person {
        hello();

        displayName;
    }

    extend(Person, {
        hello() {
            return `Hello ${this.displayName}`;
        }
        ,
        get displayName() {
                if (this.FirstName || this.LastName) {
                    if (this.FirstName && this.LastName) {
                        return this.FirstName + " " + this.LastName;
                    } else if (this.LastName) {
                        return this.LastName;
                    } else {
                        return this.FirstName;
                    }
                }

                if (this.UserName) {
                    return this.UserName;
                }

                return "N/A";
            }
    });
}

File: Base/Meta/Custom/Domain/Person.cs
namespace Allors.Meta
{
    public partial class MetaPerson
    {
        public Tree AngularHome;

        internal override void CustomExtend()
        {
            this.Delete.AddGroup("Workspace");

            this.FirstName.RelationType.AddGroup(Groups.Workspace);
            this.LastName.RelationType.AddGroup(Groups.Workspace);
            this.MiddleName.RelationType.AddGroup(Groups.Workspace);
            
            var person = this;
            this.AngularHome = new Tree(person)
                    .Add(person.Photo);
        }
    }
}